plugin.rb

```diff
@@ -497,4 +497,40 @@ def self.enqueue_assign_hooks(event, payload)
     end
   end
 
+  register_search_advanced_filter(/in:assigned/) do |posts|
+    raise Discourse::InvalidAccess unless @guardian.can_assign?
```

Comment:
How does this look from the UI? I think it might be better to make this search do nothing, rather than raise an error. What do you think?

Reply:
Yaa that's right.